### PR TITLE
[#334] Remove Compose code from template-xml

### PR DESCRIPTION
**Pull request by Wadeewee**
https://github.com/nimblehq/android-templates/issues/334

## What happened 👀

Now we have a separate template for both XML & Compose.

Remove and/or update the code accordingly, so there's no code regarding Compose left.

## Insight 📝

- Remove the config option and some libs of `Compose` in `build.gradle.kts.`
- Remove `BaseComposeFragment` and `BaseComposeFragmentCallbacks` inside the `base` directory.
- Remove directory `compose` inside directory `screen.`
- Rename directory `xml` to `home` inside the directory screen.`

Please note: We have removed some fragment extensions in `ViewModelExt` as it was unused for this time. We can consider adding them later if needed.

## Proof Of Work 📹

Screenshot_20221123_100716

### CODE REVIEW

File: template-xml/app/src/main/java/co/nimblehq/template/xml/extension/ViewModelExt.kt [hunk 1]
```diff
@@ -16,25 +14,13 @@ import androidx.lifecycle.viewmodel.CreationExtras
...
     noinline extrasProducer: (() -> CreationExtras)? = null,
     noinline factoryProducer: (() -> ViewModelProvider.Factory)? = null
 ): Lazy<VM> = OverridableLazy(viewModels(ownerProducer, extrasProducer, factoryProducer))
 
-@MainThread
-inline fun <reified VM : ViewModel> ComponentActivity.provideViewModels(
-    noinline extrasProducer: (() -> CreationExtras)? = null,
-    noinline factoryProducer: (() -> ViewModelProvider.Factory)? = null
-): Lazy<VM> = OverridableLazy(viewModels(extrasProducer, factoryProducer))
-
```
Comment (Tuubz) [inline template-xml/app/src/main/java/co/nimblehq/template/xml/extension/ViewModelExt.kt:37]: @manh-t What do you think of these changes? 

Just want to make sure these can be removed, as they are currently not being used 🙏 

Related to #137 & #139

**Review by Tuubz — APPROVED** [review]

**Comment by Tuubz** [comment]
Test

File: template-xml/app/src/main/java/co/nimblehq/template/xml/extension/ViewModelExt.kt (hunk 1 above)
Comment (Tuubz) [reply template-xml/app/src/main/java/co/nimblehq/template/xml/extension/ViewModelExt.kt:37]: Weird, this comment didn't trigger in the slack channel cc @manh-t

File: template-xml/app/src/test/java/co/nimblehq/template/xml/ui/screens/xml/HomeFragmentTest.kt [hunk 2]
```diff
@@ -6,6 +6,8 @@ import co.nimblehq.template.xml.test.TestNavigatorModule.mockMainNavigator
 import co.nimblehq.template.xml.test.getPrivateProperty
 import co.nimblehq.template.xml.test.replace
 import co.nimblehq.template.xml.ui.BaseFragmentTest
+import co.nimblehq.template.xml.ui.screens.home.HomeFragment
+import co.nimblehq.template.xml.ui.screens.home.HomeViewModel
```
Comment (luongvo) [inline template-xml/app/src/test/java/co/nimblehq/template/xml/ui/screens/xml/HomeFragmentTest.kt:10]: Please move this `HomeFragmentTest` file to `package co.nimblehq.template.xml.ui.screens.home`

image

File: template-xml/app/src/main/java/co/nimblehq/template/xml/extension/ViewModelExt.kt (hunk 1 above)
Comment (manh-t) [reply template-xml/app/src/main/java/co/nimblehq/template/xml/extension/ViewModelExt.kt:37]: @Tuubz @Wadeewee they are the handy functions for unit testing. We didn't have the test for Activity in our template, so these functions are useless. We can remove them IMO (we have a link above for reference if needed ) 🙏

Comment (luongvo) [reply template-xml/app/src/main/java/co/nimblehq/template/xml/extension/ViewModelExt.kt:37]: @Tuubz @manh-t I do think we should keep them as helper extensions except until we have another approach to do testing with Fragment and Activity. Proguard will clean unused code when doing code obfuscation. 

That's the same as having [common extensions lib](https://github.com/nimblehq/android-common-ktx/tree/develop/common-ktx/src/main/java/co/nimblehq/common/extensions) with no usage yet.

**Review by luongvo — COMMENTED** [review]
rest lgtm

File: template-xml/app/src/main/java/co/nimblehq/template/xml/extension/ViewModelExt.kt (hunk 1 above)
Comment (Wadeewee) [reply template-xml/app/src/main/java/co/nimblehq/template/xml/extension/ViewModelExt.kt:37]: @luongvo @Tuubz To make it clean when generating the script for a new project, I don't think we need to keep the unused function. We can add it later in the future if needed, Also @manh-t said 

> We didn't have the test for Activity in our template, so these functions are useless. We can remove them IMO (we have a link above for reference if needed ) 

So, I prefer to remove it for now 🙏

**Review by luongvo — APPROVED** [review]

**Review by chornerman — APPROVED** [review]
LGTM 🚀

**Pull request merged**